packages/flutter_markdown/test/text_selection_controls.dart

```diff
@@ -0,0 +1,316 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math' as math;
+
+import 'package:flutter/material.dart';
+import 'package:flutter/rendering.dart';
+
+const double _kHandleSize = 22.0;
+
+// Padding between the toolbar and the anchor.
+const double _kToolbarContentDistanceBelow = _kHandleSize - 2.0;
+const double _kToolbarContentDistance = 8.0;
+
+/// Android Material styled text selection controls.
+class MaterialTextSelectionControls extends TextSelectionControls {
```

Comment:
Why does this class need to be so elaborate when all its used for is unit tests? I'm not seeing how all of the drawing details here affect the unit test.

Reply:
I took the test from https://github.com/flutter/flutter/blob/33e261c0a2eb1f02b94c8fdfef092350c65b4d5c/packages/flutter/lib/src/material/text_selection.dart

Maybe add information about it to the test header?

Reply:
This doesn't answer my question. Why do you need an elaborate custom selection control system, with lots of code for things like the exact drawing of the selection handle, when your test doesn't set anything about those details?